src/providers/tpm_provider/utils.rs

```diff
@@ -243,6 +243,72 @@ pub fn parsec_to_tpm_signature(
     })
 }
 
+/// Validates an RSAPublicKey against the attributes we expect. Returns ok on success, otherwise
+/// returns an error.
+pub fn validate_public_key(public_key: &RSAPublicKey, attributes: &Attributes) -> Result<()> {
+    if public_key.modulus.is_negative() || public_key.public_exponent.is_negative() {
+        error!("Only positive modulus and public exponent are supported.");
+        return Err(ResponseStatus::PsaErrorInvalidArgument);
+    }
+
+    if public_key.public_exponent.as_unsigned_bytes_be() != PUBLIC_EXPONENT {
+        if crate::utils::GlobalConfig::log_error_details() {
+            error!("The TPM Provider only supports 0x101 as public exponent for RSA public keys, {:?} given.", public_key.public_exponent.as_unsigned_bytes_be());
+        } else {
+            error!("The TPM Provider only supports 0x101 as public exponent for RSA public keys");
+        }
+        return Err(ResponseStatus::PsaErrorNotSupported);
+    }
+    let key_data = public_key.modulus.as_unsigned_bytes_be();
+    let len = key_data.len();
+
+    let key_bits = attributes.bits;
+    if key_bits != 0 && len * 8 != key_bits {
```

Comment:
I'm wondering - given that the modulus is given as a vector of bytes, the assumed size is always a multiple of 8 (i.e. `len * 8`). But the size of the modulus could theoretically be something in between, like 1245, in which case the MSBits of the last byte won't be set. Now, given that the TPM will most likely reject that anyway (I think only a few values are accepted by the spec), should we reject it from here? The log messages below wouldn't be very helpful in this (rather edge) case